misk-hibernate/src/main/kotlin/misk/hibernate/TokenType.kt

```diff
@@ -0,0 +1,105 @@
+package misk.hibernate
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor
+import org.hibernate.usertype.UserType
+import java.io.Serializable
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Types
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.full.primaryConstructor
+
+/**
+ * Binds any data class that wraps a String to a varchar in MySQL. This is most useful to enforce
+ * additional type-safety on identifier columns.
+ *
+ * To use, create a data class with a single non-null `String` property. The name of this property
```

Comment:
Is non-null actually a requirement? What happens if the data type is non-null but the database value is null?

Reply:
In that case we want to get a null token, not a non-null token with a null string inside.

Ie. you get a `val GoodLuckToken?` which has a non-null string field, not a `val GoodLuckToken` that has a nullable string field.